### PR TITLE
Add output package flag

### PR DESCRIPTION
**Pull request by tony-root**
Affects both the output directory, and package statements in kt files.

### CODE REVIEW

File: kotlin/kotlin.go [hunk 1]
```diff
@@ -36,9 +39,13 @@ func(g *Generator) Generate(sources map[string]astparser.ParsedFile) map[string]
 
 	result := make(map[string][]byte, len(sources))
 	for name, file := range sources {
+		outPackage := file.Package
```
Comment (mkorolyov) [inline kotlin/kotlin.go]:
```
f := File{Package: file.Package}
if g.outPackage != "" {
  f.outPackage = g.outPackage
}
```

File: cmd/genkotlin/main.go [hunk 2]
```diff
@@ -35,15 +37,32 @@ func main() {
...
-			filePath := *outputDir + "/" + f + ".kt"
-			if err := ioutil.WriteFile(filePath, body, 0666); err != nil {
-				fmt.Fprintf(os.Stderr, "failed to save generated kotlin file %s: %v", filePath, err)
+			outPackagePath := ""
+			if *outPackage != "" {
+				outPackagePath = strings.Join(strings.Split(*outPackage, "."), "/")
+			}
+
+			// 0666 is enough, but https://github.com/golang/go/issues/22323
+			packagePath := *outputDir + "/" + outPackagePath
```
Comment (mkorolyov) [inline cmd/genkotlin/main.go]: pass direct path in `outputDir`, no need to concatenate path and create additional directories under the hood, it breaks tool API, when you specify output dir but actually files will be on other path which is confusing.

File: cmd/genkotlin/main.go [hunk 3]
```diff
@@ -35,15 +37,32 @@ func main() {
 		log.Fatalf("failed to load sources from %s excluding %s: %v", *inputDir, *excludeRegexpStr, err)
 	}
 
-	generator := kotlin.NewGenerator(kotlin.Config{UseJsonTagNames: *useJsonNames})
+	config := kotlin.Config{UseJsonTagNames: *useJsonNames}
+	if *outPackage != "" {
```
Comment (mkorolyov) [inline cmd/genkotlin/main.go]: just
```
config := kotlin.Config{
  UseJsonTagNames: *useJsonNames,
  OutPackage: *outPackage
}
```

**Review by mkorolyov — CHANGES_REQUESTED** [review]
Thanks for contribution!

File: cmd/genkotlin/main.go (hunk 2 above)
Comment (tony-root) [reply cmd/genkotlin/main.go]: So, you see it as 
`-o ~/myrepo/src/com/myapp/ -package com.myapp`
?

File: cmd/genkotlin/main.go (hunk 3 above)
Comment (tony-root) [reply cmd/genkotlin/main.go]: Done

File: cmd/genkotlin/main.go (hunk 2 above)
Comment (tony-root) [reply cmd/genkotlin/main.go]: Updated. Now we have to explicitly duplicate the package in the output path, as mentioned before.

File: kotlin/kotlin.go (hunk 1 above)
Comment (tony-root) [reply kotlin/kotlin.go]: Done

**Pull request merged**